### ISSUE TITLE
Kaitai Plugin broken, due to wrong offset

### ISSUE THREAD

**Issue by 0x61616161** [Vector35/official-plugins#1]
The Kaitai plugin does a for me unexplainable read, basically after checking the magic to select the module, but before parsing the headers multiple read operations happen, that move the file pointer from 0 offset to 0x40, at which point obviously the magic no longer exists and headers can not be mapped.

The issue seems to be in function   
from_io(cls, io):
of file kaitaistruct.py

A small rewrite of the function shows that right before the return, the offset is still at 0:
```
   @classmethod
    def from_io(cls, io):
        print('setting up stream object')
        stream = KaitaiStream(io)
        print('from_io done', io.tell(),  stream._io.tell()) # all tells == 0
        return cls(stream)
``` 

The above method is called from
def parseIo(ioObj, ksModuleName=None):
in kshelpers.py
```
try:
		ioObj.seek(0, io.SEEK_SET)
		print('after seek 0', ioObj.tell())
		# here ioObj.tell() == 0
		parsed = ksClass.from_io(ioObj)
		print('after from io', ioObj.tell()) # this prints 0x40
		ioObj.seek(0, io.SEEK_SET) # very simple, but not proper fix
		print('after reset', ioObj.tell())
		parsed._read()
	except Exception as e:
		log_error('parseIo(): kaitai module %s threw exception %s, check file type' % (ksModuleName, str(e)))
		parsed = None
```

This try block always ends up in the exception case, because it will not parse anything properly due to the 0x40 offset.

some debug output from the plugin for a test .exe
```
stream init done 0 0
tell() returning 0x0
tell() returning 0x0
from_io done 0 0
read(2) (starting at position: 0x0)
[...]
read(4) (starting at position: 0x3C)
tell() returning 0x40
after from io 64
after reset 0
```

same for elf
```
setting up stream object
tell() returning 0x0
tell() returning 0x0
stream init done 0 0
tell() returning 0x0
tell() returning 0x0
from_io done 0 0
read(4) (starting at position: 0x0)
[...]
read(2) (starting at position: 0x3E)
tell() returning 0x40
after from io 64
after reset 0
```

So those reads after the line _**from_io done 0 0**_ should probably not exist, and a workaround seems to just reset the file offset to 0.

Attached a small hello world compiled as .exe and  elf that fails.
(gcc hello_world.c -o hello_world and gcc .\hello_world.c -o .\hello_world.exe)

So basically the call to 
     parsed._read() 
fails due to the offset being not 0. Resetting of the offset via 
     ioObj.seek(0, io.SEEK_SET) 
after the line
     parsed = ksClass.from_io(ioObj)
in the file / function
     kshelpers.py  /  parseIo
seems to fix the issue.

[hello_world.zip](https://github.com/Vector35/official-plugins/files/9303590/hello_world.zip)

**Comment by lwerdna** [comment]
Thanks, we talked on slack, will hopefully get to it soon!

**Comment by lwerdna** [comment]
Fixed in Kaitai plugin version 1.1.5.
